--- libs/pcrecpp/CMakeLists.txt
+++ libs/pcrecpp/CMakeLists.txt
@@ -498,6 +498,8 @@
 ELSE(BUILD_SHARED_LIBS)
   SET(BUILD_STATIC_LIBS ON)
 ENDIF(BUILD_SHARED_LIBS)
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-format-security")
+SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-format-security")
 
 MESSAGE(STATUS "")
 MESSAGE(STATUS "")
--- libs/pcrecpp/config-cmake.h.in
+++ libs/pcrecpp/config-cmake.h.in
@@ -13,7 +13,7 @@
 #cmakedefine HAVE_MEMMOVE 1
 #cmakedefine HAVE_STRERROR 1
 #cmakedefine HAVE_STRTOLL 1
-#cmakedefine HAVE_STRTOQ 1
+//#cmakedefine HAVE_STRTOQ 1
 #cmakedefine HAVE__STRTOI64 1
 
 #cmakedefine PCRE_STATIC 1
